.github/workflows/delete-namespace.yml

```diff
@@ -15,22 +15,157 @@ permissions:
         export KUBECONFIG=$(pwd)/kubeconfig
 
+    - name: Delete all Helm-Charts in Namespace to eliminate all database connections
+      run: | 
+        # Get list of installed Helm-Charts in namespace 
+        LIST_OF_HELM_CHARTS_IN_NAMESPACE="$(helm ls \
+        --namespace ${{ inputs.namespace }} \
+        --kubeconfig $(pwd)/kubeconfig \
+        --short)"
+
+        # Check if list of installed Helm-Charts is empty 
+        if [ -z "$LIST_OF_HELM_CHARTS_IN_NAMESPACE" ]; then
+          echo "No Helm-Charts found in namespace ${{ inputs.namespace }}, skipping uninstall"
+        else
+          helm uninstall \
+            "$LIST_OF_HELM_CHARTS_IN_NAMESPACE" \
+            --namespace ${{ inputs.namespace }} \
+            --kubeconfig $(pwd)/kubeconfig \
+            --wait
+        fi
+
+    - name: Delete Keycloak instance database
+      run: | 
+        # Generate timestamp for job name uniqueness
+        DATE=$(date +%Y%m%d%H%M%S)
+
+        # Define job name for Keycloak database deletion job
+        HELM_KEYCLOAK_DELETION_JOB_NAME="psql-db-${DATE}-keycloak-${{ needs.create_ticket_nr_variable.outputs.ticket_nr }}"
+
+        # Deploy Database for Keycloak
+        helm upgrade --install \
+          "$HELM_KEYCLOAK_DELETION_JOB_NAME" \
+          ./charts/postgres_delete_databases \
+          --namespace ${{ inputs.namespace }} \
+          --set auth.existingSecret="psql-config-secret" \
+          --kubeconfig $(pwd)/kubeconfig \
+          --set database.name=${{ needs.create_keycloak_db_name.outputs.keycloak_db_name }} \
+          --wait
+
+        # Wait for Job to be completed
+        kubectl wait \
+          --namespace psql-config \
+          --kubeconfig $(pwd)/kubeconfig \
+          --for=condition=complete \
+          --timeout=120s \
+          job $HELM_KEYCLOAK_DELETION_JOB_NAME
+
+    - name: Delete SPSH instance database
+      run: | 
+        # Generate timestamp for job name uniqueness
+        DATE=$(date +%Y%m%d%H%M%S)
+
+        # Define job name for SPSH-App database deletion job
+        HELM_KEYCLOAK_DELETION_JOB_NAME="psql-db-${DATE}-database-${{ needs.create_ticket_nr_variable.outputs.ticket_nr }}"
+
+        # Deploy Database for SPSH-App
+        helm upgrade --install \
+          $HELM_KEYCLOAK_DELETION_JOB_NAME \
+          ./charts/postgres_delete_databases \
+          --namespace ${{ inputs.namespace }} \
+          --set auth.existingSecret="psql-config-secret" \
+          --kubeconfig $(pwd)/kubeconfig \
+          --set database.name=${{ needs.create_spsh_app_db_name.outputs.spsh_app_db_name }} \
+          --wait
+
+        # Wait for Job to be completed
+        kubectl wait \
+          --namespace psql-config \
+          --kubeconfig $(pwd)/kubeconfig \
+          --for=condition=complete \
+          --timeout=120s \
+          job $HELM_KEYCLOAK_DELETION_JOB_NAME
+
+    - name: Clean up database deletion Helm-Charts and its jobs from cluster
```

Comment:
are these two steps really necessary if the namespace is deleted anyway?

Reply:
Yes, the jobs are not deployed in the instance namespace because the job secret is in another namespace. This cleans up this jobs.